app/controllers/timelines_controller.rb

```diff
@@ -360,6 +353,20 @@ def load_timeline_token
       current_ability.options[:timeline_token] = @timeline_token
     end
 
+    def build_url_fragment
+      url_fragment = "noHeader=true&noFooter=true&noSourceLink=false&noVideo=false"
+
+      if @timeline.visibility == 'public' || current_user == @timeline.user
```

Comment:
I think for the public case we want the resource url fragment on the next line but the callback url on line 364 but only if the user is logged in.  A test for this case probably was missing.